### DISCUSSION TITLE
[PSD3] Exercise 5.38

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#134] [Esercizi (A-L)]
**Exercise 5.38** Add the following IEEE 754 single-precision floating-point numbers.

(a) $C0123456 + 81C564B7$
(b) $D0B10301 + D1B43203$
(c) $5EF10324 + 5E039020$

**Comment by Elia-Belli** [comment]
Convertiamo prima in formato IEEE754 single-precision (32bit), riscriviamo in "notazione scientifica", portiamo il numero con l'esponente minore alla stessa potenza dell'altro, sommiamo la mantissa in colonna. (uso ... per non riscrivere tutte le cifre della mantissa)

- (a) $C0123456 + 81C564B7$
 $C0123456 \rightarrow 1\ 10000000\ 00100100011010001010110 = -2^1 * 1.001...$
 $81C564B7 \rightarrow 1\ 00000011\ 10001010110010010110111 = -2^{-124} * 1.100... = -2^1*0.0000...$
 La mantissa del secondo addendo è costituita da soli 0 (dovuto alla traslazione della virgola), 
 quindi approssimiamo la somma a $C0123456_{16}$

- (b) $D0B10301 + D1B43203$
 $D0B10301 \rightarrow 1\ 10100001\ 01100010000001100000001 = -2^{34}*1.011... = -2^{36} * 0.01011...$
 $D1B43203 \rightarrow 1\ 10100011\ 01101000011001000000011 = -2^{36}*1.011... $
Sommando e riscrivendo in IEEE754: $1\ 10100011\ 11000000111001011000011 = D1E072C3_{16}$

- (c) $5EF10324 + 5E039020$
 $5EF10324 \rightarrow 0\ 10111101\ 11100010000001100100100 = +2^{62}* 1.111...$
 $5E039020 \rightarrow 0\ 10111100\ 00000111001000000100000 = +2^{61}* 1.000... = +2^{62}*0.1000...$
Sommando e riscrivendo in IEEE754: $1.001100101100101100110100 * 2^{63} \rightarrow 0\ 10111110\ 00110010110010110011010 = 5F19659A$

**Comment by FeddyLix17** [comment]
tutti e 3 corretti
